handlers/ffmpeg/ffmpeg.go

```diff
@@ -73,7 +75,13 @@ func (h *HandlersCollection) NewFile() httprouter.Handle {
 		// and pass the one ffmpeg gives us to UploadToOSURL instead
 		targetURLBase := reg.ReplaceAllString(job.SegmentingTargetURL, "")
 
-		if err := clients.UploadToOSURL(targetURLBase, filename, body, config.SEGMENT_WRITE_TIMEOUT); err != nil {
+		if err := backoff.Retry(func() error {
+			err := clients.UploadToOSURL(targetURLBase, filename, body, config.SEGMENT_WRITE_TIMEOUT)
```

Comment:
I'm not sure it will work, because `body` is `data io.Reader`, so I think that you cannot read from this twice.

Reply:
ah thanks good spot!

Reply:
Let's also get a test in here, since we shouldn't be relying on manually catching this stuff